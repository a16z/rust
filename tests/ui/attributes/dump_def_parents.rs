//@ normalize-stderr-test: "DefId\(.+?\)" -> "DefId(..)"
#![feature(rustc_attrs)]

fn bar() {
    fn foo() {
        fn baz() {
            #[rustc_dump_def_parents]
            || {
                //~^ ERROR: rustc_dump_def_parents: DefId
                qux::<
                    {
                        //~^ ERROR: rustc_dump_def_parents: DefId
                        fn inhibits_dump() {
                            qux::<
                                {
                                    "hi";
                                    1
                                },
                            >();
                        }

                        qux::<{ 1 + 1 }>();
                        //~^ ERROR: rustc_dump_def_parents: DefId
                        1
                    },
                >();
            };
        }
    }
}

const fn qux<const N: usize>() {}

fn main() {}
