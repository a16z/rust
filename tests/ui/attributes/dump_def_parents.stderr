error: rustc_dump_def_parents: DefId(..)
  --> $DIR/dump_def_parents.rs:8:13
   |
LL |             || {
   |             ^^
   |
note: DefId(..)
  --> $DIR/dump_def_parents.rs:6:9
   |
LL |         fn baz() {
   |         ^^^^^^^^
note: DefId(..)
  --> $DIR/dump_def_parents.rs:5:5
   |
LL |     fn foo() {
   |     ^^^^^^^^
note: DefId(..)
  --> $DIR/dump_def_parents.rs:4:1
   |
LL | fn bar() {
   | ^^^^^^^^
note: DefId(..)
  --> $DIR/dump_def_parents.rs:2:1
   |
LL | / #![feature(rustc_attrs)]
LL | |
LL | | fn bar() {
LL | |     fn foo() {
...  |
LL | |
LL | | fn main() {}
   | |____________^

error: rustc_dump_def_parents: DefId(..)
  --> $DIR/dump_def_parents.rs:11:21
   |
LL | /                     {
LL | |
LL | |                         fn inhibits_dump() {
LL | |                             qux::<
...  |
LL | |                         1
LL | |                     },
   | |_____________________^
   |
note: DefId(..)
  --> $DIR/dump_def_parents.rs:8:13
   |
LL |             || {
   |             ^^
note: DefId(..)
  --> $DIR/dump_def_parents.rs:6:9
   |
LL |         fn baz() {
   |         ^^^^^^^^
note: DefId(..)
  --> $DIR/dump_def_parents.rs:5:5
   |
LL |     fn foo() {
   |     ^^^^^^^^
note: DefId(..)
  --> $DIR/dump_def_parents.rs:4:1
   |
LL | fn bar() {
   | ^^^^^^^^
note: DefId(..)
  --> $DIR/dump_def_parents.rs:2:1
   |
LL | / #![feature(rustc_attrs)]
LL | |
LL | | fn bar() {
LL | |     fn foo() {
...  |
LL | |
LL | | fn main() {}
   | |____________^

error: rustc_dump_def_parents: DefId(..)
  --> $DIR/dump_def_parents.rs:22:31
   |
LL |                         qux::<{ 1 + 1 }>();
   |                               ^^^^^^^^^
   |
note: DefId(..)
  --> $DIR/dump_def_parents.rs:11:21
   |
LL | /                     {
LL | |
LL | |                         fn inhibits_dump() {
LL | |                             qux::<
...  |
LL | |                         1
LL | |                     },
   | |_____________________^
note: DefId(..)
  --> $DIR/dump_def_parents.rs:8:13
   |
LL |             || {
   |             ^^
note: DefId(..)
  --> $DIR/dump_def_parents.rs:6:9
   |
LL |         fn baz() {
   |         ^^^^^^^^
note: DefId(..)
  --> $DIR/dump_def_parents.rs:5:5
   |
LL |     fn foo() {
   |     ^^^^^^^^
note: DefId(..)
  --> $DIR/dump_def_parents.rs:4:1
   |
LL | fn bar() {
   | ^^^^^^^^
note: DefId(..)
  --> $DIR/dump_def_parents.rs:2:1
   |
LL | / #![feature(rustc_attrs)]
LL | |
LL | | fn bar() {
LL | |     fn foo() {
...  |
LL | |
LL | | fn main() {}
   | |____________^

error: aborting due to 3 previous errors

